### PR TITLE
update font-face mixin to allow weight and style to be set properly

### PR DESCRIPTION
**Pull request by hippogriffic**
Essentially it would appear that we've been doing this the hard way for years.
It's possible to set font-weight and font-style in the font-face mixin as long as
it's done in a specific order. Font-family must be first, then src, the font-weight
and font-style can be set.
This will mean we no longer need to use the `font` mixin to set fonts each time
we need to change the weight or style. We can simply use `font-family` to set
the web font family and then change the weight and style as needed on child elements
with font-weight and font-style

This will be a big change that will break all our old projects so please *DO NOT MERGE* until we've discussed.

@incuna/frontend 

if you're concerned about faux bolding I tested a LOT and here is proof
from chrome using this new mixin and setting font-weight and font-style:
image

from illustrator:
image

### CODE REVIEW

File: incuna-sass/mixins/_font-face.sass [hunk 1]
```diff
@@ -73,10 +60,7 @@
...
             src: font-url($filepath + ".woff") format('woff'), font-url($filepath + ".otf") format('opentype')
-
-// Use this mixin to set fonts in your project
-@mixin font($style-name, $category: sans-serif)
-    font:
-        family: '#{$style-name}', $category
-        style: normal
-        weight: normal
+        // Weight and style must be below the src declaration or will not work
+        font:
```
Comment (wytrych) [inline incuna-sass/mixins/_font-face.sass]: Do you need to repeat `font:` here?

Comment (hippogriffic) [inline incuna-sass/mixins/_font-face.sass]: sadly yes, the only way it works is if you set the weight and style after the `src` and `font-family` has to come first, slightly awkward, which is probably why we didn't figure it out until now

**Comment by pandalion** [comment]
@perry as you originally set up the font face mixin for us just wondered if you had any opinions on this/anything we're missing? :)

File: incuna-sass/mixins/_font-face.sass [hunk 2]
```diff
@@ -6,15 +6,18 @@
 
 // Use the font-face mixin to add web fonts to your projects
 // introduce the arguments in the following manner:
-// @include font-face("font-name", "font-file-name", "folder-name", "font-category")
+// @include font-face("font-family", font-weight, font-style, "font-file-name", "folder-name", "font-category")
```
Comment (pandalion) [inline incuna-sass/mixins/_font-face.sass]: how come these two don't need quotes?

File: incuna-sass/mixins/_font-face.sass [hunk 3]
```diff
@@ -73,10 +60,7 @@
...
-
-// Use this mixin to set fonts in your project
-@mixin font($style-name, $category: sans-serif)
-    font:
-        family: '#{$style-name}', $category
-        style: normal
-        weight: normal
+        // Weight and style must be below the src declaration or will not work
+        font:
+            weight: $weight
```
Comment (pandalion) [inline incuna-sass/mixins/_font-face.sass]: still not entirely sure if I get how this works, how does it select the right font file, maybe we can discuss though :)

File: incuna-sass/mixins/_font-face.sass [hunk 4]
```diff
@@ -26,40 +29,24 @@
...
-// .fancy-element
-//     @include font($font-fancy-script, cursive)
-
-// $font-formal-serif: formal-serif-font-name
-// .formal-element
-//     @include font($font-formal-serif, serif)
-
-@mixin font-face($style-name, $file, $family, $category: '', $woff2: false, $IE-8: false)
+@mixin font-face($font-family, $weight, $style, $file, $family, $category: '', $woff2: false, $IE-8: false)
     $filepath: $family + '/' + $file
```
Comment (pandalion) [inline incuna-sass/mixins/_font-face.sass]: wouldn't weight be needed in here to select eg `lato-bold.otf` as opposed to `lato.otf`?

**Comment by hippogriffic** [comment]
this should be ok to merge now if everybody agrees? it'll be a breaking change, but I've tested locally and it works beautifully

@incuna/frontend please

**Comment by pandalion** [comment]
sold on "beautifully" :D

**Comment by hippogriffic** [comment]
woo yay

**Comment by henrahmagix** [comment]
@hippogriffic Regarding the new incuna-sass font-face: I remember using that setup for eDetails years ago, but knowing we could only use it for eDetails, that WebKit was the only thing that would run them correctly. Turns out it's called [Style Linking](https://www.fontspring.com/support/troubleshooting/style-linking).

I can't seem to find any information that really corroborates that, but maybe it was due to the IE 7 & 8 restriction explained here: https://www.smashingmagazine.com/2013/02/setting-weights-and-styles-at-font-face-declaration/#style-linking
>PROBLEM: INTERNET EXPLORER 7 AND 8 CAN ONLY APPLY UP TO FOUR WEIGHTS AND STYLES TO A SINGLE FONT-FAMILY NAME LINK
>Style linking works — as long as you’re not using more than four weights and styles (and as long as the person reading your Web page isn’t the one of the one in eight people who still use Internet Explorer (IE) 7 or 8). If you use more than four weights and styles on the website, then IE 7 and 8 will convert all `light` and `medium` weights to normal, and all `black` and `heavy` weights to bold — so, you’ll lose some of your carefully set text.

So 👍  for this now

**Pull request merged**